app/main.py

```diff
@@ -1,2 +1,65 @@
+from typing import Any, Hashable
+
+
 class Dictionary:
-    pass
+    capacity = 8
+    load_factor = 2 / 3
+    size = 0
+
+    def __init__(self) -> None:
+        self.hash_table = [[None, None]] * (self.capacity + 1)
+
+    def __len__(self) -> int:
+        return self.size
+
+    def tst_ind(self, ar_ind: int) -> int:
+        ar_ind = ar_ind + 1
+        if ar_ind > self.capacity:
+            ar_ind = 0
+        return ar_ind
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        self.size += 1
+        if self.size > self.load_factor * self.capacity:
+            self.resize()
+        index = hash(key) % self.capacity
+
+        keys = [key[0] for key in self.hash_table]
+        if key in keys:
+            while keys[index] != key:
+                index = self.tst_ind(index)
+            self.hash_table[index] = [key, value]
+            self.size -= 1
+        else:
+            while self.hash_table[index] != [None, None]:
+                index = self.tst_ind(index)
+            self.hash_table[index] = [key, value]
+
+    def __getitem__(self, find_key: Hashable) -> any:
+        coli = 0
+        index = hash(find_key) % self.capacity
+        while True:
+            key_d, val = self.hash_table[index]
+            if key_d != find_key:
+                coli += 1
+                index = self.tst_ind(index)
+                if coli > self.capacity * 2:
+                    raise KeyError("Error due to big count of collisions!!")
+            else:
+                return val
```

Comment:
Can we remove `else` here and leave just `return` ? Also it is better to call return at the start and rewrite your condition to make your code less nested